app/uk/gov/hmrc/customs/inventorylinking/export/controllers/actionbuilders/AuthAction.scala

```diff
@@ -152,34 +152,20 @@ class AuthAction @Inject()(override val authConnector: AuthConnector,
         maybeEori.fold[Future[Either[Result, AuthorisedRequest[A]]]] {
           Future.successful(Left(errorResponseEoriNotFoundInCustomsEnrolment.XmlResult.withConversationId))
         } { eori =>
-
-          val maybeSubmitterHeader = maybeHeaderCaseInsensitive(XSubmitterIdentifierHeaderName)
-
-          def logAndRight(anEori: Eori): Future[Either[Result, AuthorisedRequest[A]]] = {
-            logger.debug("Authorising as non-CSP")
-            Future.successful(Right(asf.toNonCspAuthorisedRequest(anEori)))
-          }
-
-          maybeSubmitterHeader match {
-            case Some(submitterFromHeader) => maybeValidEori(maybeSubmitterHeader) match {
-              case Some(validSubmitterFromHeader) =>
-                logger.debug(s"Submitter passed in header was in a valid format: $validSubmitterFromHeader")
-                logAndRight(eori)
-              case None =>
-                logger.error(s"Submitter passed in header was in an invalid format: $submitterFromHeader")
-                Future.successful(Left(errorResponseSubmitterIdentifierHeaderInvalid.XmlResult.withConversationId))
-            }
-            case None =>
-              logger.debug("No Submitter passed in header")
-              logAndRight(eori)
+          val validated = validateSubmitterHeaderOnlyIfPresent(eori)
```

Comment:
No need for the val, place validateSubmitterHeaderOnlyIfPresent(eori) inside the if.